src/components/Form/ManageUsers.jsx

```diff
@@ -24,118 +22,169 @@ import FormSection from './FormSection';
  * @name ManageUsers
  */
 
+const WebIdEditButton = ({ lockStatus, setLockStatus }) => (
+  <IconButton
+    onClick={(e) => {
+      e.preventDefault();
+      setLockStatus(!lockStatus);
+    }}
+  >
+    {lockStatus ? <LockOpen /> : <Lock />}
+  </IconButton>
+);
+
+const formRowStyle = {
+  margin: '20px 0'
+};
+
+const textFieldStyle = {
+  margin: '8px'
+};
+
+const clearProcessing = (dispatch) => {
+  setTimeout(() => {
+    dispatch({ type: 'CLEAR_PROCESSING' });
+  }, 3000);
+};
+
+const sendUserToSolid = async (userObject, dispatch, session, state) => {
+  if (!userObject.username && !userObject.webId) {
+    runNotification(`Operation failed. Reason: No WebId provided`, 5, state, dispatch);
+    return;
+  }
+
+  dispatch({ type: 'SET_PROCESSING' });
+
+  runNotification(
+    `Adding user "${userObject.givenName} ${userObject.familyName}" to Solid...`,
+    5,
+    state,
+    dispatch
+  );
+  let listUsers = await addUserToPod(session, userObject);
+  listUsers = await getUserListActivity(session, listUsers);
+
+  runNotification(
+    `User "${userObject.givenName} ${userObject.familyName}" added to Solid`,
+    5,
+    state,
+    dispatch
+  );
+
+  clearProcessing(dispatch);
+
+  await listUsers;
+};
+
+const renderWebId = (username) => {
+  const template = ['https://', '.solidcommunity.net/profile/card#me'];
+  return `${template[0]}${username}${template[1]}`;
+};
+
 const ManageUsers = () => {
   const { session } = useSession();
   const { state, dispatch } = useStatusNotification();
   const { clearValue: clearUserGivenName, ...userGivenName } = useField('text');
   const { clearValue: clearUserFamilyName, ...userFamilyName } = useField('text');
-  const { clearValue: clearUsername, ...username } = useField('text');
+  const [username, setUsername] = useState();
+  const [webId, setWebId] = useState('');
+  const [userEditingWebId, setUserEditingWebId] = useState(false);
   const { setUserList } = useContext(UserListContext);
 
   // Event handler for adding user from users list
   const handleAddUser = async (event) => {
     event.preventDefault();
-    dispatch({ type: 'SET_PROCESSING' });
     const userObject = {
-      givenName: event.target.addUserGivenName.value,
-      familyName: event.target.addUserFamilyName.value,
-      username: event.target.addUsername.value
+      givenName: userGivenName.value,
+      familyName: userFamilyName.value,
+      username,
+      webId
     };
 
-    if (!userObject.username) {
-      runNotification(`Operation failed. Reason: No username provided`, 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
-
-    if (!userObject.givenName) {
```

Comment:
I think it's best to remove these object validations. There are jokes about how difficult it is to validate that a person's name has the correct format. Often the best thing to do is to take down whatever name they give you (regardless of its structure) and then assign them a unique ID that the system uses to separate them from others.

Also if we need robust data validations in the system, we'll probably need to set up a more object-oriented approach than this anyway. E.g. we have validations functions on the `userObject` that we call into. E.G.

```
if (!userObject.isValid()) {
  runNotification(`Operation Failed: ${userObject.getValidationErrors}`)
}
```